hilo/hilo_generator.go

```diff
@@ -0,0 +1,277 @@
+type HiLoReturnCommand struct {
+    commands.RavenCommand
+    tag string
+    last, end uint
+}
+//todo: еще нужно опреелиться с границами числовыз параметров
+func NewHiLoReturnCommand(tag string, last, end uint) (*HiLoReturnCommand, error) {
+
+    if (last < 0) || (end < 0) || (tag == "") {
+        return nil, errors.New("ArgumentOutOfRangeException")
+    }
+    ref := &HiLoReturnCommand{}
+    ref.Method = "PUT" // super(HiLoReturnCommand, self).__init__(method="PUT")
+    ref.tag = tag
+    ref.last = last
+    ref.end = end
+
+    return ref, nil
+}
+func (ref *HiLoReturnCommand)  CreateRequest(sn SrvNodes.IServerNode) {
+	ref.Url = fmt.Sprintf(`%s/databases/%s/hilo/return?tag=%s&end=%s&last=%s`, sn.GetUrl(), sn.GetDatabase(),
+		ref.tag, ref.end, ref.last)
+}
+type NextHiLoCommand struct {
+    commands.RavenCommand
+    tag, serverTag, identityPartsSeparator string
+    lastBatchSize, lastRangeMax uint
+    lastRangeAt time.Time
+}
+func NewNextHiLoCommand(tag string, lastBatchSize uint, lastRangeAt time.Time, identityPartsSeparator string, lastRangeMax uint) (*NextHiLoCommand, error) {
+    if (identityPartsSeparator == "") || (tag == "") {
+        return nil, errors.New("ArgumentOutOfRangeException")
+    }
+
+    ref := &NextHiLoCommand{
+        tag:                    tag,
+        lastBatchSize:          lastBatchSize,
+        lastRangeAt:            lastRangeAt,
+        identityPartsSeparator: identityPartsSeparator,
+        lastRangeMax:           lastRangeMax,
+    }
+    ref.Method = "GET"
+    return ref, nil
+}
+func (ref *NextHiLoCommand)  CreateRequest(sn SrvNodes.IServerNode) {
+        path := fmt.Sprintf(`hilo/next?tag=%s&lastBatchSize=%d&lastRangeAt=%s&identityPartsSeparator=%s&lastMax=%d`,
+            ref.tag, ref.lastBatchSize, ref.lastRangeAt, ref.identityPartsSeparator, ref.lastRangeMax)
+        ref.Url = fmt.Sprintf(`%s/databases/%s/%s`,sn.GetUrl(), sn.GetDatabase(), path)
+}
+//{"prefix": response["Prefix"], "serverTag": response["ServerTag"], "low": response["Low"],
+//    "high": response["High"],
+//    "last_size": response["LastSize"],
+//    "last_range_at": response["LastRangeAt"]}, err
+type nextHiloCommandJSON struct {
+    prefix string `json:Prefix`
+    serverTag string `json:ServerTag`
+    low int `json:Low`
+    high int `json:High`
+    lastSize int `json:LastSize`
+    LastRangeAt int `json:LastRangeAt`
+}
+func (ref *NextHiLoCommand) GetResponseRaw(response *http.Response) (out []byte, err error) {
+        if  response.StatusCode == 201 {
+            buf, err := ioutil.ReadAll(response.Body)
+            if err != nil {
+                return nil, err
+            }
+            //todo: по идее здесь надо сперва парсиь респонс, а потом маршалить, но я пока не нашел - где описано, как готовится ответ сервеа
+            return json.Marshal(buf)
+        }
+        if  response.StatusCode == 500 {
+            panic(errors.New(`exceptions.DatabaseDoesNotExistException(response.json(`))
```

Comment:
Why is this a panic?
Also, I'm not sure that I'm following how you are showing the url the _contents_ of the rror?